scripts/evaluation/generate_stats.jl

```diff
@@ -0,0 +1,75 @@
+using ArgParse
+using DrWatson
+@quickactivate
+using BSON
+using Random
+using FileIO
+using DataFrames
+using Base.Threads: @threads
+using GenerativeAD
+
+# pkgs which come from deserialized BSONs
+# have to be present in the Main module
+using ValueHistories
+using LinearAlgebra
+
+s = ArgParseSettings()
+@add_arg_table! s begin
+    "source_prefix"
+		arg_type = String
+		default = "experiments/images"
+		help = "Data prefix of experiment files."
+	"target_prefix"
+		arg_type = String
+		default = "evaluation/images"
+		help = "Data prefix of generated files."
+	"-f", "--force"
+    	action = :store_true
+		help = "Overwrite all generated files."
+end
+
+"""
+	generate_stats(source_prefix::String, target_prefix::String; force=true)
+
+Collects all the results from experiments in datadir prefix `source_prefix`, 
+computes evaluation metrics and stores results in datadir prefix `target_prefix` 
+while retaining the folder structure. If `force=true` the function overwrites 
+already precomputed results. 
+"""
+function generate_stats(source_prefix::String, target_prefix::String; force=true)
+	(source_prefix == target_prefix) && error("Results have to be stored in different folder.")
+	
+	source = datadir(source_prefix)
+	@info "Collecting files from $source folder."
+	files = GenerativeAD.Evaluation.collect_files(source)
+	# filter out model files
+	filter!(x -> !startswith(basename(x), "model"), files)
+	@info "Collected $(length(files)) files from $source folder."
+	# it might happen that when appending results some of the cores
+	# just go over already computed files
+	files = files[randperm(length(files))]
```

Comment:
I dont understand this.

Reply:
My suspicion is that when assigning the work to different threads, the scheduler splits the iterator into chunks and does not schedule new work once a thread completes. In case of adding files for new experiments this mean that only some of the threads were seem to be running as the rest just went through their share of files and confirmed that they exist and there is no need to compute anything. This is probably not the case, as it seems rather silly way of multi-threading, however from the CPU utilization graph it seemed to behave like that.

Reply:
Ah, now I understand. Thanks for the explanation.